webapp/src/main/java/com/box/l10n/mojito/service/scheduledjob/jobs/ScheduledThirdPartySync.java

```diff
@@ -0,0 +1,146 @@
+import com.box.l10n.mojito.service.scheduledjob.ScheduledJobRepository;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJob;
+import com.box.l10n.mojito.service.thirdparty.ThirdPartySyncJobInput;
+import com.box.l10n.mojito.utils.ServerConfig;
+import com.google.common.collect.ImmutableMap;
+import org.apache.commons.lang.text.StrSubstitutor;
+import org.quartz.DisallowConcurrentExecution;
+import org.quartz.JobExecutionContext;
+import org.quartz.JobExecutionException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Scheduled job that performs the third party sync for a repository off a cron schedule.
+ *
+ * @author mattwilshire
+ */
+@Component
+@DisallowConcurrentExecution
+public class ScheduledThirdPartySync implements IScheduledJob {
+
+  static Logger logger = LoggerFactory.getLogger(ScheduledThirdPartySync.class);
+
+  @Autowired private QuartzPollableTaskScheduler quartzPollableTaskScheduler;
+  @Autowired private ScheduledJobRepository scheduledJobRepository;
+  @Autowired private PagerDutyIntegrationService pagerDutyIntegrationService;
+
+  @Autowired ServerConfig serverConfig;
+
+  @Value(
+      "${l10n.scheduledJobs.thirdPartySync.notifications.title:MOJITO | Third party sync failed for {repository}}")
+  String notificationTitle;
+
+  private ScheduledJob scheduledJob;
+  private Long pollableTaskId;
+
+  @Override
+  public void execute(JobExecutionContext jobExecutionContext) throws JobExecutionException {
+    // Fetch the scheduled job and cast the properties
+    scheduledJob = scheduledJobRepository.findByJobKey(jobExecutionContext.getJobDetail().getKey());
+    ScheduledThirdPartySyncProperties scheduledJobProperties =
+        (ScheduledThirdPartySyncProperties) scheduledJob.getProperties();
+
+    logger.info(
+        "Third party sync for repository {} has started.", scheduledJob.getRepository().getName());
+
+    // Create ThirdPartySyncInput from scheduled job and properties
+    ThirdPartySyncJobInput thirdPartySyncJobInput =
+        new ThirdPartySyncJobInput(scheduledJob, scheduledJobProperties);
+
+    try {
+      PollableFuture<Void> task =
+          quartzPollableTaskScheduler.scheduleJobWithCustomTimeout(
+              ThirdPartySyncJob.class, thirdPartySyncJobInput, "thirdPartySync", 3600L);
+      pollableTaskId = task.getPollableTask().getId();
+      // Wait for sync to complete
+      task.get();
+    } catch (Exception e) {
+      throw new JobExecutionException(e);
+    }
+  }
+
+  @Override
+  public void onSuccess(JobExecutionContext context) {
+    logger.info(
+        "Third-Party Sync succeeded for repository {}.", scheduledJob.getRepository().getName());
+
+    // Resolve PD incident
+    pagerDutyIntegrationService
```

Comment:
I'll prefix this comment by saying it's likely for another/follow-up PR, if it will require changes.

Currently, we expect the vast majority of jobs to complete succesfully, however, for all of these jobs, when they finish, we will be doing external service calls to PD even though no incidents might exist/be relevant for PD at all.

One approach to avoid this would be to keep track of any PD incidents that we opened per jobId internally, so we know with a quick DB lookup if we ever need to attempt to issue a lot of attempts to resolve PD incidents that might never have existed in the first place.